app/models/course/material/folder.rb

```diff
@@ -8,6 +8,7 @@ class Course::Material::Folder < ActiveRecord::Base
   has_many :materials, inverse_of: :folder, dependent: :destroy, foreign_key: :folder_id,
                        class_name: Course::Material.name
   belongs_to :course, inverse_of: :material_folders
+  belongs_to :owner, polymorphic: true, inverse_of: :folder
```

Comment:
like you said, some of the fields might be repeated between the different objects, so we shouldn't use actable.